plugins/mumble_plugin_linux.h

```diff
@@ -43,7 +40,39 @@ static inline std::string readAll(std::string fn) {
 	return content;
 }
 
-static inline PTR_TYPE_CONCRETE getModuleAddr(pid_t pid, const wchar_t *modname) {
+static inline int checkProcessIs64Bit(const pid_t pid)
+{
+	// This function returns 0 if the process is 32-bit and 1 if it's 64-bit.
+	// In case of failure, it returns -1.
+	//
+	// We can know the process architecture by looking at its ELF header.
+	// The fifth byte is 1 in case the process is 32-bit or 2 in case it's 64-bit.
+	// The first 4 bytes constitute the magical number in ASCII: 0x7F 45 4c 46.
+
+	char elf[5];
+
+	std::stringstream ss;
+	ss << std::string("/proc/");
+	ss << static_cast<unsigned long>(pid);
+	ss << std::string("/exe");
+
+	std::ifstream ifs;
+	ifs.open(ss.str(), std::ifstream::binary);
+	if (ifs.is_open()) {
+		ifs.read(elf, 5);
```

Comment:
Missing check for failure. I guess in this case we could just not check and instead make sure elf is zero initialized. Bad habit to get into though.